src/controller/java/AndroidDeviceControllerWrapper.cpp

```diff
@@ -358,6 +365,13 @@ CHIP_ERROR AndroidDeviceControllerWrapper::ApplyNetworkCredentials(chip::Control
     return err;
 }
 
+CHIP_ERROR AndroidDeviceControllerWrapper::UpdateNetworkCredentials(const chip::Controller::CommissioningParameters & params)
+{
+    // this will wipe out any custom attestationNonce and csrNonce that was being used.
+    // however, Android APIs don't allow these to be set to custom values today.
+    return mAutoCommissioner.SetCommissioningParameters(params);
```

Comment:
This is a really weird approach to updating the network credentials.

As I suggested in https://github.com/project-chip/connectedhomeip/pull/20766/#discussion_r924865582 I think we should have two separate steps here: one step to do the scan and provide the delegate the scan result, and one step to ask the delegate for credentials (if any) and apply them to our existing commissioning params.

I'm probably OK with this being fixed in a followup, but it really needs to be fixed.  And if we do that, I'm not sure we need the pause/resume bit any more either.

Right now we're just creating a very complicated setup where we mostly do async operations from the commissioner and wait for responses and have some way to handle shutdown while waiting, etc (something else we are trying to add).  But then in this one case instead of using this setup we rely on out-of-band delegate notifications and the paused state...

Reply:
Created https://github.com/project-chip/connectedhomeip/issues/21314